sdk/search/azure-search/src/main/java/com/azure/search/SearchApiKeyCredential.java

```diff
@@ -11,7 +11,7 @@
  * @see SearchServiceClientBuilder
  */
 public final class SearchApiKeyCredential {
-    private final String apiKey;
+    private String apiKey;
```

Comment:
Should this be made atomic to prevent race conditions if this credential is being shared? `AtomicReference<String>`?

Reply:
@schaabs @g2vinay Any convention on this?

Reply:
Talked offline. As the api-key is at instance level. Rotate the credential for clients is thread safe.

Will leave as it is.